host_applications/linux/apps/raspicam/RaspiVid.c

```diff
@@ -1425,7 +1425,10 @@ static void encoder_buffer_callback(MMAL_PORT_T *port, MMAL_BUFFER_HEADER_T *buf
                     if(pData->pstate->frame==0)pData->pstate->starttime=buffer->pts;
                     pData->pstate->lasttime=buffer->pts;
                     pts = buffer->pts - pData->pstate->starttime;
-                    fprintf(pData->pts_file_handle,"%lld.%03lld\n", pts/1000, pts%1000);
+                    if( pData->pts_file_handle){
+                        fprintf(pData->pts_file_handle,"%lld.%03lld\n", pts/1000, pts%1000);
+                    }
+                      
```

Comment:
Yes good defensive coding, however pstate->save_pts should be 0 if not saving pts values. If the first open fails then save_pts is reset to 0 at line 2732.
Is your issue if using segmentation to switch to a new file and that fails? I think that is going to fail in a number of places, not just here. 

(Minor complaint around the whitespace before but not after in the condition. Some consistency would be nice).